flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/src/main/java/org/apache/flink/cdc/connectors/paimon/sink/v2/PaimonWriterHelper.java

```diff
@@ -111,15 +110,11 @@ private static RecordData.FieldGetter createFieldGetter(
             case TIMESTAMP_WITH_TIME_ZONE:
                 fieldGetter =
                         row ->
-                                Timestamp.fromLocalDateTime(
-                                        ZonedDateTime.ofInstant(
-                                                        row.getLocalZonedTimestampData(
-                                                                        fieldPos,
-                                                                        DataTypeChecks.getPrecision(
-                                                                                fieldType))
-                                                                .toInstant(),
-                                                        zoneId)
-                                                .toLocalDateTime());
+                                Timestamp.fromInstant(
```

Comment:
Should we need to fix Kafka/StarRocks/Doris?

https://github.com/apache/flink-cdc/blob/dd69756d3fc1e3f5261b286added5397acbd88ad/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-doris/src/main/java/org/apache/flink/cdc/connectors/doris/sink/DorisRowConverter.java#L107-L115

https://github.com/apache/flink-cdc/blob/dd69756d3fc1e3f5261b286added5397acbd88ad/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/main/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksUtils.java#L186-L195

https://github.com/apache/flink-cdc/blob/dd69756d3fc1e3f5261b286added5397acbd88ad/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-kafka/src/main/java/org/apache/flink/cdc/connectors/kafka/json/TableSchemaInfo.java#L163-L174

Reply:
Good catch, I think we should fix this issue in other connectors.

Reply:
Done fix in Paimon & Kafka connector.
For Doris & Starrocks, they  only have datetime type and don't support timestamp with local time zone, refer to https://doris.apache.org/zh-CN/docs/1.2/sql-manual/sql-reference/Data-Types/DATETIMEV2/ and https://docs.starrocks.io/docs/sql-reference/data-types/date-types/DATE/, so we need to convert timestamp_ltz to timestamp using zoneid of database, Therefore, we do not need to modify the implementation. I added tests here.
And I've picked one fix from https://github.com/apache/flink-cdc/pull/3655.